Server crash when subject is not provided on CreateStream

To reproduce:
1. Run 

````
docker run -d -p 4222:4222 -p 9292:9292 -p 8222:8222 -p 6222:6222 liftbridge/liftbridge-docker
````

2. Call "create stream" method passing only stream name.

The server outputs this and then crashes:
```
time="2019-12-23 00:03:59" level=error msg="Server failed becoming leader for partition [subject=, stream=chyRtFVDds, partition=0]: failed to subscribe to NATS: nats: invalid subject"
panic: failed to add partition to metadata store: failed to subscribe to NATS: nats: invalid subject
```

--- a/server/api.go
+++ b/server/api.go
@@ -38,6 +38,16 @@ func (a *apiServer) CreateStream(ctx context.Context, req *client.CreateStreamRe
 	a.logger.Debugf("api: CreateStream [subject=%s, name=%s, partitions=%d, replicationFactor=%d]",
 		req.Subject, req.Name, req.Partitions, req.ReplicationFactor)
 
+	if req.Name == "" {
+		a.logger.Errorf("api: Failed to create stream: name cannot be empty")
+		return nil, status.Error(codes.InvalidArgument, "Name cannot be empty")
+	}
+	// TODO: Check if valid NATS subject?
+	if req.Subject == "" {
+		a.logger.Errorf("api: Failed to create stream: subject cannot be empty")
+		return nil, status.Error(codes.InvalidArgument, "Subject cannot be empty")
+	}
+
 	var err error
 	for i := int32(0); i < req.Partitions; i++ {
 		if e := a.metadata.CreatePartition(ctx, &proto.CreatePartitionOp{

